pkg/shell/hosts_dialog.jsx

```diff
@@ -78,8 +78,14 @@ export const HostModalState = () => {
     return self;
 };
 
-export async function add_host(state) {
-    await state.show_modal({ });
+export async function add_host(state, shell_state) {
```

Comment:
Can you please add a description to "shell: Navigate to a newly added host"? TBH the code below is just magic to me, I don't know what it does. Thanks!

Reply:
> TBH the code below is just magic to me, I don't know what it does.

Yeah. This is an example of how the old code spreads complexity around, with the similar but slightly different concepts.  I'll make a new "jump_to_connection_string" function that is also used by "edit_host" and maybe elsewhere.

Reply:
This commit is mostly refactoring, but it does have a behaviour change. Can you please add a description to the commit?

Reply:
Done!